packages/gitbook-v2/src/lib/data/urls.ts

```diff
@@ -110,6 +112,26 @@ export function getURLLookupAlternatives(input: URL) {
     return { urls: alternatives, basePath, changeRequest, revision };
 }
 
+/**
+ * Get the incoming URL for a request including requests
+ * that may be proxied.
+ */
+export function getIncomingURL(
```

Comment:
I don't understand why we make it a function, instead of just fixing it in the middleware and having an e2e. As a function it's a lot more confusing to connect the dots: what does "Incomig URL" means globally?

`incomingURL` was just a variable name in the middleware, and now it's being introduced as a big thing